crates/autopilot/src/lib.rs

```diff
@@ -421,16 +439,118 @@ pub async fn main(args: arguments::Arguments) {
+        fee_discount: args.order_quoting.fee_discount,
+        min_discounted_fee: args.order_quoting.min_discounted_fee,
+        fee_factor: args.order_quoting.fee_factor,
+        liquidity_order_owners: args
+            .order_quoting
+            .liquidity_order_owners
+            .iter()
+            .copied()
+            .collect(),
+        partner_additional_fee_factors: args.order_quoting.partner_additional_fee_factors.clone(),
+    }) as Arc<dyn FeeSubsidizing>;
 
-    let mut service_maintainer = shared::maintenance::ServiceMaintenance {
-        maintainers: vec![pool_fetcher, event_updater, Arc::new(db.clone())],
+    let fee_subsidy = match cow_subsidy {
+        Some(cow_subsidy) => Arc::new(FeeSubsidies(vec![
+            fee_subsidy_config,
+            Arc::new(cow_subsidy),
+        ])),
+        None => fee_subsidy_config,
+    };
+    let mut base_estimators_instances: HashMap<_, _> = Default::default();
+    let mut get_or_create_base_estimator = move |estimator| {
+        base_estimators_instances
+            .entry(estimator)
+            .or_insert_with(|| create_base_estimator(estimator))
+            .clone()
+    };
+    let price_estimator = Arc::new(sanitized(Box::new(CompetitionPriceEstimator::new(
+        args.order_quoting
+            .price_estimators
+            .iter()
+            .map(|estimator| get_or_create_base_estimator(*estimator))
+            .collect(),
+    ))));
+    let database = Arc::new(db.clone());
+    let quoter = OrderQuoter::new(
+        price_estimator.clone(),
+        native_price_estimator.clone(),
+        gas_price_estimator,
+        fee_subsidy,
+        database.clone(),
+        chrono::Duration::from_std(args.order_quoting.eip1271_onchain_quote_validity_seconds)
+            .unwrap(),
+        chrono::Duration::from_std(args.order_quoting.presign_onchain_quote_validity_seconds)
+            .unwrap(),
+    );
+    let custom_ethflow_order_parser = EthFlowOnchainOrderParser {};
+    let onchain_order_event_parser = OnchainOrderParser::new(
+        db.clone(),
+        Box::new(quoter),
+        Box::new(custom_ethflow_order_parser),
+        DomainSeparator::new(chain_id, settlement_contract.address()),
+        settlement_contract.address(),
+    );
+    let broadcaster_event_updater = Arc::new(EventUpdater::new(
+        CoWSwapOnchainOrdersContract::new(cowswap_onchain_order_contract_for_eth_flow.clone()),
+        onchain_order_event_parser,
+        cowswap_onchain_order_contract_for_eth_flow
+            .clone()
+            .raw_instance()
+            .web3(),
+        sync_start,
+    ));
+    let maintainers: Vec<Arc<dyn Maintaining>> = if args.enable_ethflow_orders {
+        vec![
+            pool_fetcher,
+            event_updater,
+            broadcaster_event_updater,
+            Arc::new(db.clone()),
+        ]
+    } else {
+        vec![pool_fetcher, event_updater, Arc::new(db.clone())]
     };
```

Comment:
You could make it a bit easier to see which components are only required for the ethflow by initialiyzing them like this:

```suggestion
    let mut maintainers: Vec<Arc<dyn Maintaining>> =
        vec![pool_fetcher, event_updater, Arc::new(db.clone())];

    if args.enable_ethflow_orders {
        // instantiate all the things which are only needed to create the broadcaster_event_updater
        maintainers.push(broadcaster_event_updater);
    }
```